src/data/proxy/DDBPreferenceCorrelationGraph.java

```diff
@@ -108,4 +115,18 @@ public Preference getPreference(String id, PreferenceCategory category) {
             return new DDBPreferenceAdapter(item).toObject();
         }
     }
+    
+    private List<Preference> getPreferences(Page<Item, ScanOutcome> page) {
+    	Iterator<Item> iterator = page.iterator();
+    	return Stream.generate(iterator::next).map(item -> new DDBPreferenceAdapter(item).toObject()).collect(Collectors.toList());
+    }
+
+	@Override
+	public Stream<List<Preference>> batchGetPreferences(
+			PreferenceCategory category, int batchSize) {
+		Stream<Page<Item, ScanOutcome>> rawStream = StreamSupport.stream(this.preferenceTable.scan(
+				new ScanFilter(PREFERENCE_ID_ATTRIBUTE).contains(category.toString()))
```

Comment:
I agree this is kind of messy.  We can make a separate backlog item to serialize the Category separately in the DDB code (which I can take on, but would like to separate out as a separate small deliverable).  Until we do, I sort of have to keep that code.